src/mysql-util/src/schemas.rs

```diff
@@ -7,52 +7,181 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use indexmap::IndexMap;
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+
 use mysql_async::prelude::Queryable;
-use mysql_async::Conn;
+use mysql_async::{Conn, Row};
+
+use mz_repr::{ColumnType, ScalarType};
 
-use crate::desc::{MySqlColumnDesc, MySqlDataType, MySqlTableDesc};
+use crate::desc::{MySqlColumnDesc, MySqlColumnKey, MySqlTableDesc};
 use crate::MySqlError;
 
+/// Helper for querying information_schema.columns by storing the column
+/// index for easier access.
+static INFO_SCHEMA_COLS: Lazy<IndexMap<String, usize>> = Lazy::new(|| {
+    [
+        "column_name",
+        "data_type",
+        "column_type",
+        "column_key",
+        "is_nullable",
+        "ordinal_position",
+        "numeric_scale",
+        "datetime_precision",
+        "character_maximum_length",
+    ]
+    .iter()
+    .enumerate()
+    .map(|(i, col)| (col.to_string(), i))
+    .collect()
+});
+
 /// Retrieve the tables and column descriptions for tables in the given schemas.
 pub async fn schema_info(
     conn: &mut Conn,
-    schemas: Vec<&str>,
+    schemas: &Option<Vec<String>>,
 ) -> Result<Vec<MySqlTableDesc>, MySqlError> {
-    // Get all tables of type 'Base Table' in schema
-    let table_q = "SELECT table_name, table_schema
-                   FROM information_schema.tables
-                   WHERE table_type = 'BASE TABLE'
-                   AND table_schema IN (?)";
-    let table_rows: Vec<(String, String)> = conn.exec(table_q, schemas).await?;
+    let table_rows: Vec<(String, String)> = if let Some(schemas) = schemas {
+        // Get all tables of type 'Base Table' in specified schemas
+        assert!(!schemas.is_empty());
+        let table_q = format!(
+            "SELECT table_name, table_schema
+                       FROM information_schema.tables
+                       WHERE table_type = 'BASE TABLE'
+                       AND table_schema IN ({})",
+            schemas.iter().map(|_| "?").join(", ")
+        );
+        conn.exec(table_q, schemas).await?
+    } else {
+        // Get all tables in non-system schemas.
+        // TODO(roshan): Many users create user-defined tables in the `mysql` system schema, since mysql doesn't
```

Comment:
A Notion task/GH issue for this will help ensure it gets documented. Btw. Is there any documentation task/issue?

Reply:
@joacoc good point -- I'll plan to add this and other TODOs in these PRs as tasks in our notion doc today https://www.notion.so/materialize/MySQL-Source-Prototype-8838b49eb2124e3580a4e9922761d423

>  Btw. Is there any documentation task/issue?
There is not yet! I can create that too, though this is currently a 'prototype' feature so there is no urgency at the moment for comprehensive docs.